pkg/http-handler/http-handler.go

```diff
@@ -75,6 +76,34 @@ func getNamespaceAndCluster(req *http.Request) (string, string) {
 	return requestNamespace, requestCluster
 }
 
+// GetHeaderNamespaces returns a list of namespaces from the header request
+// The name and the value of the header field is specified by 2 environment variables:
+// - KUBEAPPS_NAMESPACE_HEADER is a name of the expected header field,
+//     e.g. KUBEAPPS_NAMESPACE_HEADER=X-Consumer-Groups
+// - KUBEAPPS_NAMESPACE_PATTERN is a regular expression and it matches only single regex group,
+//     e.g. KUBEAPPS_NAMESPACE_PATTERN=namespace:([\w-]+)
+func getHeaderNamespaces(req *http.Request) []string {
+	headerName := os.Getenv("KUBEAPPS_NAMESPACE_HEADER")
+	if headerName == "" {
+		return []string{}
+	}
+	headerPattern := os.Getenv("KUBEAPPS_NAMESPACE_PATTERN")
+	if headerPattern == "" {
+		return []string{}
+	}
+	r := regexp.MustCompile(headerPattern)
+	headerNamespacesOrigin := strings.Split(req.Header.Get(headerName), ",")
+	var namespaces = []string{}
+	for _, n := range headerNamespacesOrigin {
+		rns := r.FindStringSubmatch(strings.TrimSpace(n))
+		if len(rns) < 2 {
```

Comment:
Why are you checking `< 2` like this?

Reply:
The first position in the slice returns the entire string, matching groups start from position no 1.
However, `FindStringSubmatch` returns `nil` when no match found, which I added to the condition.
Thanks for bringing my attention.